### ISSUE TITLE
Postgres 10: not seeing any tables on the side

### ISSUE THREAD

**Issue by adamJLev** [TablePlus/TablePlus#2162]
Moving issue to the right project here, from the windows one: https://github.com/TablePlus/TablePlus-Windows/issues/61#issuecomment-717341421

1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):
PostgreSQL 10.12 
I havent tried other versions.

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):
Version 3.10.0 (348)

3. The steps to reproduce this issue:
- Open connection to postgresql
- No tables on the side

By the way, when I run the query that I found on another ticket here that the maintainer suggested, I do see the right tables show up (I blurred them out of the screenshot, but they are there):

<img width="1233" alt="Screen_Shot_2020-10-27_at_5_07_56_PM" src="https://user-images.githubusercontent.com/188466/97329406-c2895900-1855-11eb-9499-ac3cbf05e9dc.png">

**Comment by huyphams** [comment]
Hi @adamJLev could you please run this query and let me know the result?

```
(SELECT table_name, table_schema, table_type FROM information_schema.tables) UNION (SELECT c.relname AS table_name, n.nspname AS table_schema, 'MATERIALIZED VIEW' FROM pg_catalog.pg_class c JOIN pg_namespace n ON n.oid = c.relnamespace WHERE c.relkind = 'm');
```

Thank you!

**Comment by adamJLev** [comment]
@huyphams yeah thats the same query you can see in my second screenshot above.

It returns all the tables in the database successfully, including all from the  `public`  schema - but the tables dont show up in the sidebar for some reason. any clue?

**Comment by huyphams** [comment]
Hmm, could you double-check if there is any extra whitespaces in the `table_schema` result of the query above? (double click the field to check)

**Comment by adamJLev** [comment]
@huyphams No whitespaces or anything weird that I can see, only letters and underscores.
I can send you the exact output privately if that helps in debugging, maybe by email? let me know

**Comment by renatonitta** [comment]
Hi @adamJLev, check the user privileges.

I was having the same issue and figure it out by changing the user.

User with limited privileges to the selected database:
![Screen Shot 2020-12-10 at 12 37 36](https://user-images.githubusercontent.com/24940/101795185-6701ea80-3ae6-11eb-9636-4c29790f1174.png)

User with all privileges to the selected database:
![Screen Shot 2020-12-10 at 12 38 53](https://user-images.githubusercontent.com/24940/101795430-a6303b80-3ae6-11eb-8d1e-37339c40feac.png)

In my case I changed the user, but you can try alter user privileges.

**Comment by adamJLev** [comment]
Thanks @renatonitta I think that would work unfortunately I cant get more access on this db (company thing). 

It does seem like a bug somewhere though, because the SQL query that @huyphams provided above does return all the tables as expected  - but they dont show up on the panel for some reason.

**Comment by mirestrepo** [comment]
I am running into the same issue in Postgres12. My user privileges are `CTc`... I'm wondering if any of you were able to find a work around?

**Comment by mirestrepo** [comment]
Actually in my case `\dt` shows all the tables but 
```
(SELECT table_name, table_schema, table_type FROM information_schema.tables WHERE table_schema = 'public') UNION (SELECT c.relname AS table_name, n.nspname AS table_schema, 'MATERIALIZED VIEW' FROM pg_catalog.pg_class c JOIN pg_namespace n ON n.oid = c.relnamespace WHERE c.relkind = 'm');
```
does not....

**Comment by jakequade** [comment]
Been over a year, any plans on looking into this?

**Comment by adamJLev** [comment]
yeah, I'm still having this issue...

**Comment by huyphams** [comment]
I think this is a permission issue, not really a TablePlus issue, however, in the upcoming review I have split this query
(SELECT table_name, table_schema, table_type FROM information_schema.tables WHERE table_schema = 'public') UNION (SELECT c.relname AS table_name, n.nspname AS table_schema, 'MATERIALIZED VIEW' FROM pg_catalog.pg_class c JOIN pg_namespace n ON n.oid = c.relnamespace WHERE c.relkind = 'm');

Into two separated queries, so if the user only has read information_schema.tables permission, can load tables.

**Comment by adamJLev** [comment]
that sounds like a great idea, thanks @huyphams  !

**Comment by FelipeCO14** [comment]
Facing the same issue... And while it could be some sort of permission issue, the tables show just fine in other database tools like DBeaver, so I believe there is definitely some work to be done on the tool side as well...

**Comment by bpannes** [comment]
I agree with @FelipeCO14 and I am using Postgres 11.16 here

**Comment by bpannes** [comment]
> that sounds like a great idea, thanks @huyphams !
Did you manage to make it work?

**Comment by adamJLev** [comment]
It still does not work for me using latest version (5.1.0), even though its been a while and I assume @huyphams has gotten that proposed change in by now.

I still dont see any tables on the sidebar for some databases.

@huyphams Actually I just tested that query above and it does run successfuly, and I can see all my tables in there. but the sidebar is still empty, what gives

**Comment by huyphams** [comment]
Have you tried to change the schema at the bottom  of the left sidebar? @renatonitta

**Comment by adamJLev** [comment]
@huyphams yes public schema. See attached 

![Notification_Center](https://user-images.githubusercontent.com/188466/201648233-3dc68732-1306-433b-a212-e2a4ce63968a.png)

**Comment by keithharvey** [comment]
Same issue for me using Redshift.

**Comment by MouhamedMokkhtar** [comment]
Same here !!!!!!

**Comment by huyphams** [comment]
Can you show me a screenshot @MouhamedMokkhtar ?
